### ISSUE TITLE
[Bug]: TS error when using remix 1.7.5 and fetcher

### ISSUE THREAD

**Issue by pobidowski** [airjp73/remix-validated-form#205]
### Which packages are impacted?

- [X] `remix-validated-form`
- [ ] `@remix-validated-form/with-zod`
- [ ] `@remix-validated-form/with-yup`
- [ ] `zod-form-data`

### What version of these packages are you using?

- `@remix/react`: 1.7.5
- `remix-validated-form`: 4.6.4

### Steps to Reproduce the Bug or Issue

```
import { ValidatedForm } from 'remix-validated-form';

...
const fetcher = useFetcher<SomeType>();

...
return (
    <ValidatedForm
      ...
      fetcher={fetcher}
    >
      ...
    </ValidatedForm>
);
```

Error:
```TS2322: Type 'FetcherWithComponents<SerializeObject<UndefinedToOptional<SomeType>>>' is not assignable to type 'FetcherWithComponents<never> | undefined'.```

### Expected behavior

You can use fetcher without TS error.

### Platform

- macOS

**Comment by kwiat1990** [comment]
I'm facing the same error.

**Comment by stephen776** [comment]
Same here

**Comment by zolzaya** [comment]
Any updated?

**Comment by airjp73** [comment]
Should be fixed now in 4.6.5. Please feel free to re-open if it's still not working!

**Comment by stephen776** [comment]
Still having the same issue on latest. I'll provide an example asap. I'm passing `fetcher` as a prop into my component which renders the form.

**Comment by kwiat1990** [comment]
Thanks for quick response and fix to the issue! After updating to `4.6.5` I don’t have the get the error anymore, so for me it has worked.
